### PR TITLE
[PL-26239]: fix for list response

### PR DESCRIPTION
**Pull request by bhavya181**
response schema for listUserRepositries and listAppnstallations was different, and that's why it was throwing error, so changes the output.
Tested with writing integartion test with my credentials, now, its working fine.

### CODE REVIEW

**Review by adivishy1 — APPROVED** [review]

File: scm/driver/github/repo.go [hunk 1]
```diff
@@ -14,7 +14,7 @@ import (
 	"github.com/drone/go-scm/scm"
 )
 
-type repository struct {
+type Repository struct {
```
Comment (mohitg0795) [inline scm/driver/github/repo.go:17]: From what I understand, it makes the struct public. Why do we need to make it public if all references are from same package/file?
Sorry, I may be naive here, just trying to understand its impact

Comment (bhavya181) [reply scm/driver/github/repo.go:17]: json unmarshal and marshal only works when name of struct start with capital letter

Comment (mohitg0795) [reply scm/driver/github/repo.go:17]: Oh interesting. Thanks for the info.

**Review by mohitg0795 — APPROVED** [review]

File: scm/driver/github/repo.go (hunk 1 above)
Comment (tphoney) [reply scm/driver/github/repo.go:17]: image
https://github.com/drone/go-scm/pull/219
repositories should be lower case as @mohitg0795 said

**Review by tphoney — CHANGES_REQUESTED** [review]

File: scm/driver/github/repo.go [hunk 2]
```diff
@@ -53,6 +53,11 @@ type hook struct {
 	} `json:"config"`
 }
 
+type RepositoryListResponse struct {
```
Comment (tphoney) [inline scm/driver/github/repo.go:56]: should start with a lowercase letter

**Review by tphoney — CHANGES_REQUESTED** [review]

File: scm/driver/github/repo.go (hunk 1 above)
Comment (bhavya181) [reply scm/driver/github/repo.go:17]: but then, RepositoryListResponse will not be able to map to Repository struct as it does not unmarshal that struct

Comment (bhavya181) [reply scm/driver/github/repo.go:17]: Can you suggest any other method, so that it will map correctly

Comment (tphoney) [reply scm/driver/github/repo.go:17]: can you show me where it doesnt work, the unit tests passed and it unmarshalled successfully

if you look at the do method for all drivers, it marshalls / unmarshalls without all of the structs. The other structs all start with a lowercase letter.

File: scm/driver/github/repo.go (hunk 2 above)
Comment (bradrydzewski) [reply scm/driver/github/repo.go:56]: agree this should be lowercase.   Also we do not suffix with `Response` so it should be `repositoryList` instead

**Comment by bradrydzewski** [comment]
The `repository` struct should never be exposed as public, it should be kept private and converted to `scm.Repository`.  The purpose of this library is to convert vendor-specific structures to common abstractions.  The `scm.Repository` is the common abstract that must be used.

If the response structure is different, you should create a new `applicationRepositoryList` structure instead.  Then you would convert the repositories in the `applicationRepositoryList` to a `[]scm.Repository` structure.

File: scm/driver/github/repo.go (hunk 1 above)
Comment (bhavya181) [reply scm/driver/github/repo.go]: ya, I have changed them now

File: scm/driver/github/repo.go (hunk 2 above)
Comment (bhavya181) [reply scm/driver/github/repo.go]: changed

**Review by tphoney — APPROVED** [review]

**Pull request merged**